### DISCUSSION TITLE
Returns different result from eslint

### DISCUSSION THREAD

**Discussion by strowk** [rizowski/eslint-watch#169] [General]
### Environment

  - Node Version: v12.19.0
  - Eslint-Watch Version: Eslint-Watch: v7.0.0
  - Eslint Version: v7.15.0

### Basic Description of the problem

If I run `eslint  ./src --ext .js --ext .ts --ext .tsx`, it gives output 
```
✖ 1 problem (0 errors, 1 warning)
```

but esw says "Clean"
```
 esw -w  ./src --ext .js --ext .ts --ext .tsx 
✓ Clean (5:36:27 PM)
```

### How to reproduce it

  1. Command: `esw -w  ./src --ext .js --ext .ts --ext .tsx`
  2. should give same output as `eslint  ./src --ext .js --ext .ts --ext .tsx`

### Debug output:

```
  esw:eslint Executing [ '--ext', [ '.js', '.ts', '.tsx' ], './src' ] +0ms
  esw:eslint {
  esw:eslint   command: 'eslint --ext .js,.ts,.tsx ./src',
  esw:eslint   exitCode: 0,
  esw:eslint   stdout: '',
  esw:eslint   stderr: '',
  esw:eslint   all: undefined,
  esw:eslint   failed: false,
  esw:eslint   timedOut: false,
  esw:eslint   isCanceled: false,
  esw:eslint   killed: false
  esw:eslint } +1s
✓ Clean (5:41:46 PM)
```

**Comment by rizowski** [comment]
This is by design. There have been previous suggestions for turning it off. What is your use case?

**Reply by strowk** [reply]
Sorry, I'm confused. It's designed to return "Clean", when in fact eslint found a problem? But what's the goal to have it running then?

**Reply by rizowski** [reply]
Yes, Eslint doesn't return a failed state on warnings even though it output's them as problems. The resulting exit code is 0. Because warnings are optional fixes, Eslint-watch will report it as "Clean" because of that. Unless something has changed in Eslint that I'm unaware of. I know some projects like create-react-app force eslint to fail on warnings and usually when eslint is running on a CI environment but that's not default behavior.

Am I understanding your concern correctly?

**Reply by strowk** [reply]
Aah, I see. Warnings are not ok though, from perspective of CI (I think). Yes, because it's CRA. 

However `eslint` still prints warnings to output. Don't you think that `esw` should repeat that behavior? Or at least document that it ignores warnings?

In documentation ( https://github.com/rizowski/eslint-watch ) I see `--quiet Report errors only - default: false`. 
Sounds like warnings should still be reported by default? 

Additional note:
Personally I'm running eslint and esw locally because I don't want to wait for CI to fail on it. Whether it reports me errors or warnings I don't really care, because CI does not. 

I understand when for example `yarn start` (from CRA) does not fail on warnings and still runs, because warnings are normal somewhere mid-process and I'd like to ignore them while I'm stil trying different stuff. 
But once I'm in "lint fixing" stage, I want to know everything that will fail on CI without waiting to run it and that's the part where I'd like to look at esw output and one by one kick out everything that does not look good. 
And the fact, when esw silently drops warnings seems counterintuitive to me in general and particulary so in my scenario.

I'm not sure how everyone else is using esw though, maybe that's too specific case... 
Does that make sense?

**Reply by rizowski** [reply]
@strowk Sorry for the delay, I'm currently away. I think your arguments are fair. Originally when I built this I went with assumptions that the company I worked for, at the time, had. This shouldn't be too hard to fix.

I would be fine with changing it so ESW reports warnings and does not display clean unless the `--quiet` flag is passed. When I get back and available I will start looking into this.
